src/test/java/com/VetApp/PurrgentCare/services/PetServiceImplementationTest.java

```diff
@@ -100,83 +110,80 @@ public void deletePet_whenValidInput() {
     @Test
     public void getAllPets_withValidInput_returnsAllPets() {
         // given
-        final var fakeCountOfFakePets = fakeDataGenerator.generateRandomInteger();
-        final var expected = fakeDataGenerator.generatePetList(fakeCountOfFakePets);
-        given(mockPetRepository.findAll())
-                .willReturn(expected);
+        final var fakePets = fakeDataGenerator.generatePetList();
+        final List<PetResponse> fakePetResponses = fakeDataGenerator.generateFakePetResponses();
+
+        given(mockPetRepository.findAll()).willReturn(fakePets);
+        when(mockMapper.map(any(Pet.class), eq(PetResponse.class)))
+                .thenAnswer(invocationOnMock -> {
+                    Pet fakePet = invocationOnMock.getArgument(0);
+                    int index = Arrays.asList(fakePets).indexOf(fakePet);
+                    return fakePetResponses.get(index);
+                });
 
         // when
         final var actual = serviceUnderTest.getAllPets();
 
+
         // then
-        then(actual).isEqualTo(expected);
-        then(actual).hasSize(fakeCountOfFakePets);
+        then(actual).isEqualTo(fakePetResponses);
+        then(actual).hasSize(fakePetResponses.size());
     }
 
     @Test
     public void getAllPets_whenNoPets_returnsEmptyList() {
         // given
-        final var expected = fakeDataGenerator.generatePetList(0);
-        given(mockPetRepository.findAll())
-                .willReturn(expected);
+        final var fakePets = new ArrayList<Pet>();
+        final List<PetResponse> fakePetResponses = fakeDataGenerator.generateFakePetResponses();
```

Comment:
shouldn't this be an empty response?

Reply:
I'll set the PetResponse to an empty ArrayList like the Pets above them.